src/utils.ts

```diff
@@ -1405,3 +1419,32 @@ export function parsePackageVersion({ version }: { version: string }): {
   const [major, minor, patch] = version.split('.').map((n: string) => Number.parseInt(n, 10));
   return { major, minor, patch };
 }
+
+/**
+ * Fisher–Yates Shuffle
+ *
+ * Reference: https://bost.ocks.org/mike/shuffle/
+ * @param sequence - items to be shuffled
+ * @param limit - Defaults to `0`. If nonzero shuffle will slice the randomized array e.g, `.slice(0, limit)` otherwise will return the entire randomized array.
+ */
+export function shuffle<T>(sequence: Iterable<T>, limit = 0): Array<T> {
+  const items = Array.from(sequence); // shallow copy in order to never shuffle the input
+
+  if (limit > items.length) {
+    throw new MongoRuntimeError('Limit must be less than the number of items');
+  }
+
+  let remainingItemsToShuffle = items.length;
+  while (remainingItemsToShuffle !== 0) {
```

Comment:
can we optimize this very slightly by changing this to `> 1` ? Technically, on that last pass, we have already shuffled all we can, right? So the `randomIndex` can only be `0`, and remaining placement can only be `0`

Reply:
follow up question, if this does in fact generate a random sorting of the array, wouldn't the backward sorting be as random as the forward sorting? since after we go though `x` iterations of the loop, the last `x` elements will never be changed, so if we have a limit, say `n`, couldn't we stop after `n` iterations and return the corresponding slice from the end of the array?

Reply:
We can do the `> 1` change but I think we run into an issue if we try and shuffle some subset. Our test that does a limit of 1, 5 times over, and expects at least 2 unique values started failing. Because now, among the "large" input array it's very unlikely that we get the chance to shuffle the first item out of place, such that the concluding slice returns something different.